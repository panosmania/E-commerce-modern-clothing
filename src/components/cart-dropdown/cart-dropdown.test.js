import { shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";

import { CartDropdown } from "./cart-dropdown.component";
import CustomButton from "../custom-buttom/custom-buttom.component";
import CartItem from "../cart-item/cart-item.component";

describe("CartDropdown component", () => {
  let wrapper;
  let mockHistory;
  let mockToggleCartHidden; //0
  const mockCartItems = [{ id: 1 }, { id: 2 }, { id: 3 }]; //2

  beforeEach(() => {
    mockHistory = {
      push: jest.fn(),
    };
    mockToggleCartHidden = jest.fn();

    const mockProps = {
      cartItems: mockCartItems,
      history: mockHistory,
      toggleCartHidden: mockToggleCartHidden, //3
    };

    wrapper = shallow(<CartDropdown {...mockProps} />);
  });

  it("should render CartDropdown component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call history.push when button is clicked", () => {
    wrapper.find(`[id="counter"]`).simulate("click");
    expect(mockHistory.push).toHaveBeenCalled();
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });

  it("should render 1 <CustomButton /> element", () => {
    expect(wrapper.find(CustomButton)).toHaveLength(1);
    //expect(wrapper.find("CustomButton")).toHaveLength(1); δουλευει και με "" απο οτι καταλαβαινω τα component καλυτερα να τα βαεις χωρις "" καθως με "" δεν δουλευει παντα
  });

  it("should render an equal number of CartItem components as the cartItems prop", () => {
    expect(wrapper.find(CartItem).length).toEqual(mockCartItems.length);
  });

  it("should render EmptyMessageContainer if cartItems is empty", () => {
    const mockProps = {
      cartItems: [],
      history: mockHistory,
      toggleCartHidden: mockToggleCartHidden, //4
    };

    const newWrapper = shallow(<CartDropdown {...mockProps} />);
    expect(newWrapper.exists(".empty-message")).toBe(true);
  });
});
