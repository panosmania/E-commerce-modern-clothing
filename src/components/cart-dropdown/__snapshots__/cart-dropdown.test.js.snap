// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown component should render CartDropdown component 1`] = `
<div
  className="cart-dropdown"
>
  <div
    className="cart-items"
  >
    <Memo(CartItem)
      item={
        Object {
          "id": 1,
        }
      }
      key="1"
    />
    <Memo(CartItem)
      item={
        Object {
          "id": 2,
        }
      }
      key="2"
    />
    <Memo(CartItem)
      item={
        Object {
          "id": 3,
        }
      }
      key="3"
    />
  </div>
  <CustomButton
    id="counter"
    onClick={[Function]}
  >
    GO TO CHECKOUT
  </CustomButton>
</div>
`;
